pyuvdata/uvdata/uvdata.py

```diff
@@ -578,6 +672,362 @@ def _set_flex_spw(self):
+        Set multi_object to True, and adjust required paramteres.
+
+        This method is typically not be called directly by users; instead it is called
+        by the file-reading methods to indicate that an object has multiple sources
+        together in the same data set.
+
+        Parameters
+        ----------
+        preserve_source_info : bool
+            Preserve the source information located in object_name, and for phased data
+            sets, also phase_center_ra, phase_center_dec, phase_center_epoch and
+            phase_center_frame. Default is True.
+
+        Raises
+        ------
+        ValueError
+            if the telescope_name is not in known telescopes
+        """
+        self.multi_object = True
+
+        # Mark once-option arrays as now required
+        self._object_id_array.required = True
+        self._Nobjects.required = True
+        self._object_dict.required = True
+        self._object_name.form = ("Nobjects",)
+        self._object_name.strict_type = False
+
+        # This should technically be required for any phased data set, but for now,
+        # we are only gonna make it mandatory for multi-obj data sets.
+        self._phase_center_app_ra.required = True
+        self._phase_center_app_dec.required = True
+        self._phase_center_app_pa.required = True
+
+        if preserve_source_info:
+            if self.phase_type == "phased":
+                # "Phased" objects in the parlance of multi-object is a "sidereal"
+                # source -- a celestial body that does not require an ephemeris to
+                # describe the motion of.
+                self.object_name = [self.object_name]
+                object_dict = {
+                    "object_type": "sidereal",
+                    "object_lon": self.phase_center_ra,
+                    "object_lat": self.phase_center_dec,
+                    "coord_frame": self.phase_center_frame,
+                    "coord_epoch": self.phase_center_epoch,
+                }
+                if object_dict["coord_frame"] is None:
+                    object_dict["coord_frame"] = "icrs"
+            elif self.phase_type == "drift":
+                # A "drift" phase-typed is really just an unphased data set, not
+                # actually what one might call a drift scan (array is co-planar to a
+                # fixed az and el). In the lingo of multi-object data sets, we just
+                # call this unphased.
+                self.object_name = ["unphased"]
+                # Unphased dictionary are simple -- they just require two elements
+                object_dict = {"object_type": "unphased"}
+
+            elif self.phase_type == "unknown":
+                raise ValueError(
+                    "Cannot preserve object properties of the existing data set if "
+                    "the phasing type is unknown! Either specify phasing type, or "
+                    "set /preserve_source_info=False."
+                )
```

Comment:
So I think the only two ways currently to get the unknown phased type are through a corner-case in MIRIAD (which may actually go away), and if a dataset was previously recorded as "unknown" type in UVH5. I'd personally be in favor of getting rid of it, since an unknown phasing scheme means that you can't really do any operations on the data, but perhaps this is something that needs discussion.

Reply:
@plaplant agrees! I will get rid of it.

Reply:
Alright, unknown phase type has been removed -- both both cases I mentioned above, it seems like the typical case was where `phase_type="drift"` but some information (e.g., telescope location) doesn't quite match up, so I've set `phase_type="drift"` in lieu of `phase_type="unknown"` in those cases (and to throw a warning that one of these corner cases was detected).